README.md

```diff
@@ -1 +1,682 @@
-# messaging-with-activemq
\ No newline at end of file
+# messaging-with-activemq
+
+This Guide will illustrate how to configure ballerina services as a JMS producer (one way messaging, aka fire and forget mode) and JMS consumer with ActiveMQ message broker.
+
+Let's consider a real world secenario where online order management system. Clients can place their orders, then Order accepting ballerina service will place that orders into a message broker queue, then Order dispatcher ballerina service will route them to a difference queues by considering the message content( it will check retail order or whole sale order), then respective ballerina services will consume the messages from each queue.
+
+
+![alt text](https://github.com/tdkmalan/messaging-with-activemq/blob/master/JMS_bal_Service.png)
+
+
+# The following are the sections available in this guide.
+
+- What you'll build
+- Prerequisites
+- Implementation
+- Testing
+- Deployment
+- Observability
+
+# Prerequisites
+
+- Ballerina Distribution
+- Apache ActiveMQ 5.12.0
+
+**Note -**
+After installing the JMS broker, copy its .jar files into the <BALLERINA_HOME>/bre/lib folder
+For ActiveMQ 5.12.0: Copy activemq-client-5.12.0.jar, geronimo-j2ee-management_1.1_spec-1.0.1.jar and hawtbuf-1.11.jar
+
+A Text Editor or an IDE
+
+
+# Implementation
+If you want to skip the basics, you can download the git repo and directly move to the "Testing" section by skipping "Implementation" section.
+
+# Create the project structure
+
+Ballerina is a complete programming language that supports custom project structures. Use the following package structure for this guide.
+
+```
+messaging-with-activemq
+ └── guide
+      ├── order_accepting_service
+      │    ├── order_accepting_service.bal
+      │    └── tests
+      │         └── order_accepting_service_test.bal
+      │ 
+      │── order_dispatcher_service
+      │    └── order_dispatcher_service.bal
+      │   	
+      └── retail_order_process_service
+      │    └── retail_order_process_service.bal
+      │ 
+      └── retail_order_process_service
+	       └── wholesale_order_process_service.bal	
+```
+     
+- Create the above directories in your local machine and also create empty .bal files.
+- Then open the terminal and navigate to messaging-with-activemq/guide and run Ballerina project initializing toolkit.
+
+```
+  $ ballerina init
+```
+# Developing the service
+
+Let's get start with the implementation of the "order_accepting_service.bal", which acts as a http endpoint which accept request from client and publish messages to a JMS destination. "order_dispatcher_service.bal" process the each message recieve to the Order_Queue and route orders to the destinations queues by considering their message content. "retail_order_process_service.bal" and "wholesale_order_process_service.bal" are listner services for the retail_Queue and Wholesale_Queue.
```

Comment:
spelling double check, sentence..

"Let's get start with the implementation of the "order_accepting_service.bal", which acts as http endpoint which accepts the request from the client and publishes messages to a JMS destination. "order_dispatcher_service.bal" process each message received to the Order_Queue and route orders to the destinations queues by considering their message content. "retail_order_process_service.bal" and "wholesale_order_process_service.bal" are listener services for the retail_Queue and Wholesale_Queue."

Reply:
Grammer and Spelling verified and corrected